src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2222,8 +2222,10 @@ private bool getReadonlyStaticFieldValue(CORINFO_FIELD_STRUCT_* fieldHandle, byt
             FieldDesc field = HandleToObject(fieldHandle);
             Debug.Assert(field.IsStatic);
 
-            if (!field.IsThreadStatic && field.IsInitOnly && field.OwningType is MetadataType owningType)
+            if (!field.IsThreadStatic && !filed.HasRva && field.IsInitOnly && field.OwningType is MetadataType owningType)
             {
+                Debug.Assert(valueOffset == 0); // is only used for RVA (not supported currently)
```

Comment:
Do you plan to implement this as part of this PR? It should not be difficult.

Reply:
Right, but currently this path is not hit and I'm trying to figure out why the IR shape is completely different and I can't get FieldSeq pointer in NativeAOT case.